### ISSUE TITLE
Hardware donations

### ISSUE THREAD

**Issue by SirCmpwn** [SirCmpwn/wlroots#171]
We would appreciate hardware donations to help some of our developers build support for as many devices as we can. Currently we would find these useful:

- HiDPI displays
- Displays that use DisplayPort MST
- Drawing tablets that offer mode switching on the tablet pad

Email sir@cmpwn.com if you'd like to donate any hardware.

**Comment by SirCmpwn** [comment]
I have received a $50 donation with the stated intent of purchasing HiDPI hardware for testing on wlroots.

**Comment by SirCmpwn** [comment]
I have purchased a 4K display.

**Comment by maletor** [comment]
Can this be closed or do you think you'll need to test a 5K monitor too ;)

**Comment by ascent12** [comment]
I think some 10-bit and/or YUV capable hardware would be helpful.

**Comment by SirCmpwn** [comment]
And I could use a drawing tablet with more capabilities than mine, too.

**Comment by braderhart** [comment]
Do you still need drawing tablets, and what is the cheapest you'd need to get this working?

**Comment by SirCmpwn** [comment]
Cheap probably isn't feasible for the drawing tablet we need. Support for cheap drawing tablets is fleshed out and working well, the main issue is fancy drawing tablets with mode switching and ring controls and crazy shit. Something like this could get us pretty far: http://a.co/fF75wSn

 Could also just get in touch with Wacom and ask if they'd be willing to give away any hardware in exchange for adding support for them.

**Comment by braderhart** [comment]
> Cheap probably isn't feasible for the drawing tablet we need.
True that, but cheap for what I was thinking is under $600. Let me know if there is something else that has the requirements for testing to help get this closed out.

> Could also just get in touch with Wacom and ask if they'd be willing to give away any hardware in exchange for adding support for them.
Would you want me to reach out? In the meantime, depending on how important this is to anyone else, I wouldn't mind contributing for the hardware.

**Comment by SirCmpwn** [comment]
>Would you want me to reach out?

Sure, if you have the connections that'd be great.

>In the meantime, depending on how important this is to anyone else, I wouldn't mind contributing for the hardware.

It's more about how important it is to you, I guess. I want to support drawing tablets well on wlroots (and Sway) but I'm not sure how much the drawing tablet audience (mainly artists, I guess?) intersects with the wlroots-compositor-user audience (mainly geeks, famed for their artistic inability).

**Comment by SirCmpwn** [comment]
My reason for wanting drawing tablet support is so I can play [osu!](https://osu.ppy.sh/), and that use-case is already well supported :smile:

**Comment by braderhart** [comment]
There may be some other higher priority items, but precision drawing support is really important to me as well. I want to try Inkscape and Krita tablet support, and see what other applications exists. I'm sort of envisioning sway compositor as being a good replacement for Android eventually, or at least that is my hopes.

**Comment by Ongy** [comment]
https://bugs.kde.org/show_bug.cgi?id=379174 towards kirta... Fell free to bug them a little. They are under the KDE umbrella and kwin put X11 under maintainance mode, so there may be some argument.

Drawing tablets "should" work
I have tested pen, buttons on pen, buttons on pad and strips. I could test erasor but don't think I did.

We have an implementation for rings, but that's largely untested, since nobody has the hardware atm.
What's unimplemented is mode-switching. Tbh. I'm not even 100% sure what that is about, but I think the tablet linked earlier is the cheapest that has it.

**Comment by AerysBat** [comment]
@Ongy Please do not bug Krita about this. Complaining is not going to make Wayland support happen any faster, and it will not increase the number of krita devs who are interested in doing this, which is currently zero. They already spend too much time putting out fires in buggy libraries and Wayland is another flaming dumpster at the moment. Despite Kwin's X11 feature freeze, it remains the default for KDE. 

If you want this to happen in 2018 maybe you could write a GSoC proposal.

**Comment by ReK2Fernandez** [comment]
haha add to the list a steam account to run linux steam games :)

**Comment by braderhart** [comment]
@SirCmpwn 
> It's more about how important it is to you, I guess. I want to support drawing tablets well on wlroots (and Sway) but I'm not sure how much the drawing tablet audience (mainly artists, I guess?) intersects with the wlroots-compositor-user audience (mainly geeks, famed for their artistic inability).
It is important to me, but I hope that it helps the community and I believe that it will.  I just saw your new update to your Hackathon regarding [Purism](https://puri.sm/) and I see other opportunities. Optimized code and a properly configure Vim with patched fonts is art to me.

**Comment by SirCmpwn** [comment]
Thanks to @braderhart for hooking me up with a nicer drawing tablet. When it arrives I'll update this issue with details on its capabilities.

**Comment by SirCmpwn** [comment]
Drawing tablet showed up - thanks again! I have confirmed that it supports rings and will add support for them to wlroots accordingly. However, it does not support modes. I'm still not sure how "modes" even translate to marketing material, it will be hard to find one of these.

**Comment by SirCmpwn** [comment]
Clarification: it does support modes! It does not seem to support "mode groups", whatever that is. Anyway, very good tablet, has a large spread of capabilities to improve support for. Thanks @braderhart!

**Comment by SirCmpwn** [comment]
I added two more calls for donations:

- [ ] [freedreno compatible hardware](https://github.com/freedreno/freedreno/wiki#devices) for YUV support
- [ ] VR headset (that can be driven using only open source software/drivers)

**Comment by braderhart** [comment]
@SirCmpwn This is the one I'm familiar with, but may be a bit dated:

http://www.osvr.org/hdk2.html

Will that work for your efforts?

**Comment by SirCmpwn** [comment]
Yes, that'll do nicely. But let someone else have the pleasure of donating, too :smile:

**Comment by emersion** [comment]
We could also make use of one of these things:

* A [FreeSync](https://en.wikipedia.org/wiki/FreeSync) monitor
* A multi-GPU setup that supports GPU hotplugging, e.g. via Thunderbolt (https://github.com/swaywm/wlroots/issues/1278)

**Comment by emersion** [comment]
In addition to the FreeSync/VRR/Adaptive Sync monitor which is blocking https://gitlab.freedesktop.org/wayland/wayland/issues/84, we could also use a HDR-capable monitor (those are pretty expensive) for color management.

**Comment by Algram** [comment]
I've emailed Lenovo asking about one of their displaylink docking stations as a hardware donation. Maybe this will ease the implementation of https://github.com/swaywm/wlroots/issues/1823 since the person can test on real hardware.

I hope this is fine for you.

**Comment by nstickney** [comment]
@emersion Is there any particular FreeSync monitor someone wants? Or should we just get the cheapest one we can? I would like to donate towards that one.

**Comment by emersion** [comment]
Sorry about the delay. I just noticed that my GPU is AMD GCN 1.1 and doesn't support FreeSync, which makes things more complicated.

**Comment by nstickney** [comment]
@emersion No worries... it looks like an RX570 can be had for about $120, or an RX 550 for $100. Is that a useful way to spend the money?

https://pcpartpicker.com/products/video-card/#A=2&sort=price&page=1

**Comment by emersion** [comment]
I'm now getting stable income, so I'm considering throwing some money at a new screen and/or GPU. To maximize the number of DRM features I could work on, the best would be to have a FreeSync and HDR-capable hardware. It seems like a Polaris (RX 400/500) would have support for both. I looked at my local second-hand exchange website, and the prices are around 100 euros too. Regarding the screen, it seems BenQ offers cheaper HDR screens than other vendors (something like EW3270U or EW2780 would do). Otherwise, could always go for a non-HDR screen, any FreeSync-capable one would do.

**Comment by nstickney** [comment]
@emersion :+1:
Let me know how to send the money. Merry Christmas!

**Comment by emersion** [comment]
Sent a message on Mastodon with details. Thanks!

**Comment by josh-lucidd** [comment]
I'd donate 50$ towards FreeSync/VRR implementation. Doesn't matter if the money goes into a Monitor or into a GPU.

**Comment by ddevault** [comment]
I have received a $104 donation for the purpose of developing GPU hotplugging support.

**Comment by zachlatta** [comment]
I'm happy to pitch in $100 for GPU hotplugging support. What's the best way to send the money?

**Comment by ddevault** [comment]
I can accept that donation at https://drewdevault.com/donate (leave a note with the purpose of the donation), but I'm not sure if we have any devs lined up who would be interested in taking on that work should the hardware be provided to them.

**Comment by zachlatta** [comment]
Donation made! That's OK. I hope someone will pop up at some point :-).
Thank you for Sway. It's completely transformed how I use Linux.

--
Zach Latta
https://zachlatta.com <http://zachlatta.com>
@zachlatta <https://twitter.com/zachlatta> / fb
<https://www.facebook.com/crynix> / github <https://github.com/zachlatta>

**Comment by jakobrs** [comment]
Regarding
>     * [ ]  A device that doesn't have a render engine, such as DisplayLink
There's [the `gud` DRM driver](https://github.com/notro/gud). It lets you use a Raspberry Pi 4 as a display adapter (without rendering). The host driver is upstreamed in Linux 5.13.
